docker/binder/start.hive

```diff
@@ -38,21 +40,84 @@ fi
 
 export HIVE_HOME=$HIVE_PARENT_DIR/$HIVE_FOLDER_NAME
 
-# Create hive warehouse folder
-mkdir $HIVE_WAREHOUSE_DIR
-
 # Copy the needed configs to Hive folder
 cp $RESOURCE_DIR/hive/config/hive-site.xml ${HIVE_HOME}/conf/
 
 # Set Hive warehouse path in the hive-site.xml
 sed -i.bak "s~HIVE_WAREHOUSE_DIR~$HIVE_WAREHOUSE_DIR~g" ${HIVE_HOME}/conf/hive-site.xml
 
+# Check for Java 8 + 11 for tox (also in /notebooks/tests/scripts/start_hive)
+if [[ -z ${JAVA8_HOME} || -z ${JAVA11_HOME} || ! -d ${JAVA8_HOME} || ! -d ${JAVA11_HOME} ]] ; then
+  cat <<! > /dev/stderr
+
+
+============================================================================================================
+Define the JAVA8_HOME and JAVA11_HOME environment variables to point to Java 8 and Java 11 development kits.
+============================================================================================================
+
+Need Java 8 for Hive server to work.
+Java 11 (not newer!) is required for Spark, but also Nessie.
+
+
+!
+  exit 1
+fi
+
+# Kill an already running hiveserver
+if [[ -f $HIVE_PID_FILE ]] ; then
+  kill "$(cat $HIVE_PID_FILE)" || true
+  rm $HIVE_PID_FILE
+fi
+
+# Remove an already metastore-db
+if [[ -d $HIVE_DB ]] ; then
+  echo "Removing existing $HIVE_DB"
+  rm -rf $HIVE_DB
+fi
+
+# (Re-)create hive warehouse folder
+rm -rf $HIVE_WAREHOUSE_DIR
+mkdir -p $HIVE_WAREHOUSE_DIR
+
 # Initialize Hive's Derby database
 $HIVE_HOME/bin/schematool -dbType derby -initSchema
 echo "Finished initializing Derby database for Hive."
 
 # increase the Heap memory being used by Hive-MapReduce jobs
 export HADOOP_HEAPSIZE=1500
 
+# Use Java 8 for Hive :facepalm:
```

Comment:
that is really annoying

Reply:
After 5+ years of Java 11 and 6+ years after Java 9, yea.